hoomd/hpmc/GPUHelpers.cuh

```diff
@@ -33,12 +45,14 @@ __device__ inline unsigned int computeParticleCell(const Scalar3& p,
         kb = 0;
 
     // identify the bin
-    if (f.x >= Scalar(0.0) && f.x < Scalar(1.0) && f.y >= Scalar(0.0) && f.y < Scalar(1.0) && f.z >= Scalar(0.0) && f.z < Scalar(1.0))
+    if (!strict || (f.x >= Scalar(0.0) && f.x < Scalar(1.0) &&
```

Comment:
What does this accomplish? When strict is false, out of bounds ib, jb, or kb values will result in returns values wrapping to other cells in the list and possibly cells outside of the addressable range.

Reply:
`computeParticleCell()` is now used in more than one place, in the move proposal kernel as well as in the overlap kernels. 
See here https://github.com/glotzerlab/hoomd-blue/blob/770703b87e8f0a5aa4e6c083124cd12764325192/hoomd/hpmc/IntegratorHPMCMonoGPUMoves.cuh#L110-L111 and here https://github.com/glotzerlab/hoomd-blue/blob/770703b87e8f0a5aa4e6c083124cd12764325192/hoomd/hpmc/IntegratorHPMCMonoGPUMoves.cuh#L149-L150
In these examples, the `strict=true` argument is passed, as opposed to the places where it's being used in the overlap kernels, e.g. in
https://github.com/glotzerlab/hoomd-blue/blob/770703b87e8f0a5aa4e6c083124cd12764325192/hoomd/hpmc/IntegratorHPMCMonoGPUDepletants.cuh#L204-L205

The reason for the distinction is that an particle that is out of the box generates a return value for the bin (`0xffffffff`) that is always out of bounds, a condition that can therefore easily be intercepted, as above. The moves kernel then sets the `d_reject_out_of_cell` flag for that particle, which is checked by the overlaps kernel and if set, allows it to skip any out of bounds memory accesses to the cell list. The `false` argument is not really necessary (passing true would be perfectly fine), it's intended to be more of a statement for anyone who reads the code that at that point we rely on the cell being valid.

Reply:
Alright, that makes sense. The place where `false` is passed doesn't mind if it gets an in-bounds value for a particle that is out of the box.